common/concurrency/ConcurrentTask.h

```diff
@@ -0,0 +1,108 @@
+#ifndef SORBET_CONCURRENTTASK_H
+#define SORBET_CONCURRENTTASK_H
+
+#include "absl/synchronization/barrier.h"
+#include "common/ConstExprStr.h"
+#include "common/Counters.h"
+#include "common/concurrency/ConcurrentQueue.h"
+#include "common/concurrency/WorkerPool.h"
+
+namespace sorbet {
+template <typename I, typename WorkerState, bool CollectCounters = false> class ConcurrentTask {
+    std::shared_ptr<ConcurrentBoundedQueue<I>> inputq;
```

Comment:
these are pretty heavy templates, you may want to pimpl to save compilation time. your call